### PR TITLE
Cover letter print times updates with functional test

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Improve functional tests for producer](https://tools.hmcts.net/jira/browse/RPE-277)

### Change description ###

adding single test suite to check updates on `sent_to_print_at` and `printed_at`.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    private AuthTokenValidator tokenValidator;
+
+    @MockBean
+    private QueueClientSupplier queueClientSupplier;
+
+    @Mock
+    private IQueueClient queueClient;
+
+    @Value("${status-update-service-name}")
+    private String serviceName;
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java]: perhaps `serviceAllowedToMakeUpdates` for clarity in tests?

Comment (doncem) [reply src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java]: W ill it be sufficient if I add a dockblock instead? Making vars so long seems a bit unnecessary. But docblock is useful

File: src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    @Test
+    public void should_return_204_after_updating_letter_in_db() throws Exception {
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java]: you test much more than that

File: src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+        updateLetterStatus(UUID.randomUUID(), "sent_to_print_at")
+            .andExpect(status().isNotFound());
+        updateLetterStatus(UUID.randomUUID(), "printed_at")
+            .andExpect(status().isNotFound());
+    }
+
+    private ResultActions updateLetterStatus(UUID letterId, String field, ZonedDateTime time) throws Exception {
+        MockHttpServletRequestBuilder request =
+            put("/letters/" + letterId.toString() + "/" + field.replaceAll("_", "-"))
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java]: why not just pass string with `-`?

Comment (doncem) [reply src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java]: the body contains `_`. one or another

File: src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java (hunk 2 above)
Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java]: perhaps 'should update letter in db'? status codes are tested in unit tests

Comment (doncem) [reply src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java]: probably followed the pattern. can add `_and_verify_the_changes`

**Pull request merged**